actionpack/lib/abstract_controller/caching/fragments.rb

```diff
@@ -136,7 +136,7 @@ def expire_fragment(key, options = nil)
 
       def instrument_fragment_cache(name, key) # :nodoc:
         payload = instrument_payload(key)
-        ActiveSupport::Notifications.instrument("#{name}.#{instrument_name}", payload) { yield }
+        ActiveSupport::Notifications.instrument("#{name}.#{instrument_name}".freeze, payload) { yield }
```

Comment:
Just confirmed what you said, I will open an PR and change it back.
But I think Ruby should also support this behavior?